opensrs/xcp.py

```diff
@@ -117,34 +119,34 @@ def __init__(self, action, object, attributes=None, timeout=None, **kw):
         self.timeout = timeout
 
     def get_content(self):
-        return self.ops_message.to_string()
+        return self.ops_message.to_string().encode('utf-8')
 
     def sign(self, private_key):
-        firstpass = hashlib.md5(self.get_content() + private_key).hexdigest()
-        return hashlib.md5(firstpass + private_key).hexdigest()
+        firstpass = hashlib.md5((self.get_content() + private_key)).hexdigest()
+        return hashlib.md5(firstpass.encode('utf-8') + private_key).hexdigest()
 
 
 class XCPChannel(object):
     def __init__(self, host, port, username, private_key, default_timeout):
         self.host = host
         self.port = port
         self.username = username
-        self.private_key = private_key
+        self.private_key = private_key.encode('utf-8')
```

Comment:
I don't think it's reasonable to assume that the `private_key` will be a Unicode string, when run under Python 2.